### DISCUSSION TITLE
Do I really need a Filter Box in my dashboard if I want to open it using "preselect_filters"?

### DISCUSSION THREAD

**Discussion by r-lomba** [apache/superset#13368] [General]
Hi all, the final goal I'm trying to achieve is:

- Expose a pre-filtered, single Slice to the public
- Ideally, I'd open the URL of the Dashboard containing the Slice to be exposed using the **preselect_filters** parameter (example follows) to have the resulting Dashboard ready to be exposed in an iframe e.g. using the desired time range
- But it looks like I can't do it **without including a Filter Box in my Dashboard**. And I don't want to **present a Filter Box to the end user**! I just want to present the meaningful Slice/Chart

For example, if I have a Dashboard **made up of a Filer Box and a Slice**, using the following URL allows me to have the results correctly pre-filtered:

`http://<THE_SERVER>:8081/superset/dashboard/27/?preselect_filters={"178": {"name": ["John Smith"], "__time_range": "Last week"}}`

The problem is that I'll also have a Filter Box in full sight, and I don't want to present it in the final embedding iframe

But, obviously, **if I remove the Filter Box element** from Dashboard "27", it doesn't work anymore. The Dashboard that will be generated using the above URL won't be filtered at all

The question:

- Given the above, how to present a pre-filtered Slice/Chart to the end user, without presenting him/her a Dashboard that includes a Filter Box?
- Maybe there's a way to "hide" the Filter Box element inside a Dashboard, so that I can present the whole Dashboard anyway?
- Or maybe there's a way to include in the calling URL the fact that I want to "maximize" the resulting Slice of interest?
- Or, finally, there's a way to pre-filter a Dashboard that doesn't include a Filter Box, and I'm just not aware of it?

Thanks for any useful insights!

**Comment by durchgedreht** [comment]
Hi,
178 should be the slice_id the filter is applied to. So you need to change this ID to whatever ID your representation-only slice has and then it should work. You can also use a slice without dashboard and embed that if this suits your use case better.

In case your Slice with representation has ID 123 use:
http://<THE_SERVER>:8081/superset/dashboard/27/?preselect_filters={"123": {"name": ["John Smith"], "__time_range": "Last week"}}

Without having the dec source to this assumption (anybody please correct otherwise), preselect_filters addressed to the filter_box will be propagated to all slices unless filters are not set on the ignore list.

**Reply by r-lomba** [reply]
Hi and thank you for your reply. I have to say, when I started working on this in the beginning i was making the **very same** assumptions as you are kindly explaining! But I soon found out that:

- Regarding the "_You can also use a slice without dashboard and embed that if this suits your use case better_" assumption, unfortunately no, it doesn't work. You need a dashboard containing the slice you want to expose! That is, if we take the practical example made in the previous posts, you need the "**27**" ID, otherwise you simply can't build a dynamic URL to be called pointing at your chart... You can't just do "**.../dashboard/?preselect_filters={"123"...**"
- More than that, I discovered that even if you build a dashboard and you put a slice in it (e.g. slice "**123**") but you don't put a filter box element as well into the dashboard, containing all the fields you wish to pass via the "**preselect_filters**" parameter in the dynamic URL, when you call the URL the dashboard comes up with no errors, but **it just doesn't get filtered!**

So: I promise I have already tried everything you suggest, but no cigar. Hence I decided to write this question... It really looks that what the dynamic URL does is to pass the "**preselect_filters**" parameters to the filter box that **has** to be present in the dashboard, and then it looks like the filter box delivers filter events to the desired slice

I agree with you: this is so strange! **Why is then an "123" parameter even necessary**, if it's the filter box delivering the filter events to whatever slice is already present on the dashboard?

I believe this is a strong inconsistency il the logic of this mechanism, but I hoped to find an answer to my original post anyway....

Any thoughts and/or practical workarounds?

Thanks!

**Comment by durchgedreht** [comment]
Ok, I have to admit there was some steps and infos missing. But it's doable. It's just not very obvious. Easiest way to understand is as follows:

1.  go into the explore view (go to menu entry "Charts")
2. select your chart of interest
3. add the filters you ant here (just to understand what's going on)
4. Have a look into the URL and put it into an URLdecoder

You'll notice there's all parameters injected using encoded JSON. Inside the **url_params** there's a **slice_id**. Also you'll find  **adhoc_filters**, which you can change to your needs. If you'd like to have the Chart render without all controls, just add _'&standalone=true'_ to the end of the URL. You also might not need all the params listed, so adjust to your needs. Hope this helps.

Sidenote: I found export/import lacks the ability to deal with charts that do not belong to a dashboard. So I ended up putting all those charts into a fake dashboard, so those can be exported and imported easily. Also you want them to have only the same datasource in one chart to avoid an import mess. So best use one fake dashboard per database and you'll be able to export and import your work easily. Charts alone do not offer that functionality (if, please let me know ;-)

**Reply by r-lomba** [reply]
Thanks. I have tried extensively following your suggestions, I understand the principles you described and to some extent I got good results already... It's not easy, though. Also, "can explore" permission is involved, and I didn't want to leave this accessible to a "Public" user.

Nevertheless, your help has been really appreciated, and thanks to you I now have the instruments to proceed with my trials. I'll keep this thread posted if I succeed in finding a satisfactory enough solution, for the posterity :)

**Comment by Ibby-B** [comment]
Have you considered, hiding the chart in the dashboard with CSS? if all you want to do is hide it, that should be simple enough.

**Comment by tce-darshan** [comment]
Hi all,

I need to pass dynamic ids from URL. I tried adding filter box and preselect_filters in the URL like below but it is not working. 
`https://<domain>/superset/dashboard/16/?preselect_filters={“237”: “user_oid": “abc” }`
Any help is appreciated. Thank you :)
